packages/okta-react-native/android/src/main/java/com/oktareactnative/OktaSdkBridgeModule.java

```diff
@@ -0,0 +1,462 @@
+import com.okta.oidc.net.response.IntrospectInfo;
+import com.okta.oidc.net.response.UserInfo;
+import com.okta.oidc.storage.SharedPreferenceStorage;
+import com.okta.oidc.util.AuthorizationException;
+
+public class OktaSdkBridgeModule extends ReactContextBaseJavaModule implements ActivityEventListener {
+
+    private final ReactApplicationContext reactContext;
+    private OIDCConfig config;
+    private WebAuthClient webClient;
+
+    public OktaSdkBridgeModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+        this.reactContext.addActivityEventListener(this);
+    }
+
+    @Override
+    public String getName() {
+        return "OktaSdkBridge";
+    }
+
+    @ReactMethod
+    public void createConfig(String clientId,
+                             String redirectUri,
+                             String endSessionRedirectUri,
+                             String discoveryUri,
+                             ReadableArray scopes,
+                             Promise promise
+    ) {
+
+        try {
+            String[] scopeArray = new String[scopes.size()];
+
+            for (int i = 0; i < scopes.size(); i++) {
+                scopeArray[i] = scopes.getString(i);
+            }
+
+            this.config = new OIDCConfig.Builder()
+                    .clientId(clientId)
+                    .redirectUri(redirectUri)
+                    .endSessionRedirectUri(endSessionRedirectUri)
+                    .scopes(scopeArray)
+                    .discoveryUri(discoveryUri)
+                    .create();
+
+            this.webClient = new Okta.WebAuthBuilder()
+                    .withConfig(config)
+                    .withContext(reactContext)
+                    .withStorage(new SharedPreferenceStorage(reactContext))
+                    .setRequireHardwareBackedKeyStore(true) // TODO: remember to set it to true when releasing SDK
+                    .create();
+
+            promise.resolve(true);
+        } catch (Exception e) {
+            promise.reject(OktaSdkError.OKTA_OIDC_ERROR.getErrorCode(), e.getLocalizedMessage(), e);
+        }
+    }
+
+    @ReactMethod
+    public void signIn(final Promise promise) {
+        try {
+            Activity currentActivity = getCurrentActivity();
+
+            if (currentActivity == null) {
+                promise.reject(OktaSdkError.NO_VIEW.getErrorCode(), OktaSdkError.NO_VIEW.getErrorMessage());
+                return;
+            }
+
+            if (webClient == null) {
+                promise.reject(OktaSdkError.NOT_CONFIGURED.getErrorCode(), OktaSdkError.NOT_CONFIGURED.getErrorMessage());
+                return;
+            }
+
+            final SessionClient sessionClient = webClient.getSessionClient();
+            final WritableMap params = Arguments.createMap();
+
+            webClient.registerCallback(new ResultCallback<AuthorizationStatus, AuthorizationException>() {
```

Comment:
The callback can be triggered without the
 
```java
webClient.signIn(currentActivity, null);
``` 
being called. An edge case where the ReactActivity -> WebBrowser. ReactActivity is destroyed so the callback is broken. But the OIDC SDK will keep track of the browser results so whenever the callback is registered again, the status is returned immediately. So will webClient.signIn be called if the promise is resolved?

Reply:
Resolved